library/src/main/java/com/chuckerteam/chucker/internal/data/repository/HttpTransactionDatabaseRepository.kt

```diff
@@ -4,11 +4,13 @@ import androidx.lifecycle.LiveData
 import com.chuckerteam.chucker.internal.data.entity.HttpTransaction
 import com.chuckerteam.chucker.internal.data.entity.HttpTransactionTuple
 import com.chuckerteam.chucker.internal.data.room.ChuckerDatabase
+import com.chuckerteam.chucker.internal.data.room.HttpTransactionDao
 import com.chuckerteam.chucker.internal.support.distinctUntilChanged
 
-internal class HttpTransactionDatabaseRepository(private val database: ChuckerDatabase) : HttpTransactionRepository {
+internal class HttpTransactionDatabaseRepository(database: ChuckerDatabase) :
+    HttpTransactionRepository {
 
-    private val transactionDao get() = database.transactionDao()
+    private val transactionDao: HttpTransactionDao = database.transactionDao
```

Comment:
Why this change?

Reply:
Removed val on the `database` in consctructor, transactionDao can be initiallized directly when constructing.

Reply:
Sorry, I don't get this response. Which constructor and which initialization do you have on mind? DAOs are lazy initialized and synchronized in Room.

Reply:
![image](https://user-images.githubusercontent.com/10363352/112725243-31c4bf80-8f52-11eb-81bf-752a5f46b6f6.png)
Here, `database` is just a injected param only provide `dao` for `transactionDao`

Reply:
Ah, thanks. Didn't catch that. Why would it be a problem to keep a reference to the database instead of DAO?

Reply:
`database` just used to provide `dao` for `transactionDao`, we can remove the reference after constructor for min visibility.

Reply:
Visibility is still the same. `HttpTransactionDatabaseRepository` has access to database anyway. More meaningful would be to pass DAO as a constructor parameter.

Reply:
Yeah, it's only duty is as a constructor parameter, should not let other functions to access.